nucypher/policy/conditions/ecdsa.py

```diff
@@ -41,30 +45,27 @@ def validate_message(self, value):
         @validates("signature")
         def validate_signature(self, value):
             if not is_context_variable(value):
-                # Try to decode it to ensure it's valid base64
                 try:
-                    base64.b64decode(value)
+                    bytes.fromhex(value)
                 except Exception as e:
                     raise ValidationError(
-                        f"Invalid signature format, must be base64 encoded: {str(e)}"
+                        f"Invalid signature format, must be hex encoded: {str(e)}"
                     )
 
         @validates("verifying_key")
         def validate_verifying_key(self, value):
             try:
-                VerifyingKey.from_pem(value.encode())
+                bytes.fromhex(value)
```

Comment:
Was the reason for not trying to create the VerifyingKey the fact that you now need to specify the curve to create the VerifyingKey?

If so, then you can check multiple schema values at the same time using `@validates_schema` instead of `@validates` - see something analogous for `ContractCall` https://github.com/nucypher/nucypher/blob/main/nucypher/policy/conditions/evm.py#L273.

Reply:
✔️